lightning/src/routing/router.rs

```diff
@@ -38,7 +38,8 @@ pub struct RouteHop {
 	/// The channel_announcement features of the channel that should be used from the previous hop
 	/// to reach this node.
 	pub channel_features: ChannelFeatures,
-	/// The fee taken on this hop. For the last hop, this should be the full value of the payment.
+	/// The fee taken on this hop (for paying for the use of the *next* channel in the path).
```

Comment:
Let's say the path is our_node->Alice->Bob->Carol->Dave.
So, the fee taken on Alice->Bob channel is actually paying for the use of the Bob->Carol transfer  — this is how LN works.
(Is this correct?).

I find this a bit counter-intuitive, that's why I expanded the comment.

I'm happy to take more suggestions how to clarify it even furthr

Reply:
Huh. I'm not sure what the right answer is, but this does seem to contradict the spec? from Bolt 7: "After a channel has been initially announced, each side independently announces the fees and minimum expiry delta it requires to relay HTLCs through _this_ channel." (emphasis mine).

Reply:
I don't think there's a contradiction. It requires to relay through *this* channel, yes, but how it charges the fee is another question :)

Let's try a simple example.
Imagine this
Alice 100<>100 Bob 100<>100 Carol

Alice tries to send 10 to Carol via Bob, Bob is gonna charge 5 in fees.

After the payment, the state will be:
Alice 85<>115 Bob 90 <> 110 Carol

So, you can see, although Alice pays for the use of Bob-Carol channel, the Bob-Carol channel is not even aware of the fees Bob charged, they are taken care of on the previous hop.
This is what I try to communicate here.
cc @ariard ?

Reply:
Ah, I see what you're saying! I wouldn't be opposed to including that example in the comment^ But It might just be me who's confused lol, so I'm fine to leave it as is.